src/templates_parser.adb

```diff
@@ -3557,17 +3557,18 @@ package body Templates_Parser is
                -- Rewrite --
                -------------
 
-               procedure Rewrite (T : in out Tree) is
+               procedure Rewrite (T : Tree) is
+                  L : Tree := T;
```

Comment:
I don't understand the use of a temporary `Tree` object?

Reply:
I don't want to change T (was in out) as we do change this pointer (see Move_To_Last). And this was so changing the pointer of if true/false sub-tree:

```
                           Rewrite (L.N_True);
                           Rewrite (L.N_False);
```

So changing the actual tree, which was fully wrong.